File: web/src/types/store.ts
export interface PricingTier {
  table_type: string;
  price: number;
  unit: string;
  description?: string;
}

export interface MemberCard {
  name: string;
  price: number;
  duration?: string;
  benefits?: string[];
}

export interface OperationProfile {
  facility_features?: string[];
  pricing_features?: string[];
  coach_features?: string[];
  tournament_features?: string[];
  member_features?: string[];
  atmosphere_features?: string[];
  service_features?: string[];
  marketing_features?: string[];
  target_audience?: string[];
}

export interface StoreCreate {
  name: string;
  city?: string | null;
  district?: string | null;
  address?: string | null;
  phone?: string | null;
  business_hours?: string | null;
  table_count?: number | null;
  table_types?: string | null;
  pricing?: PricingTier[] | string | null;
  member_cards?: MemberCard[] | string | null;
  operation_profile?: OperationProfile | null;
  has_private_room?: boolean;
  has_coaching?: boolean;
  has_tournament?: boolean;
  has_parking?: boolean;
  target_customers?: string | null;
  style?: string | null;
  brand_style?: string | null;
  advantages?: string | null;
  common_activities?: string | null;
  // Expanded profile fields
  coach_count?: number | null;
  coach_service_types?: string | null;
  coach_price_range?: string | null;
  cue_price_range?: string | null;
  table_brands?: string | null;
  cue_brands?: string | null;
  other_equipment?: string | null;
  membership_types?: unknown | null;
  recharge_rules?: unknown | null;
  membership_benefits?: unknown | null;
  daily_avg_customers?: number | null;
  peak_hours?: string | null;
  avg_spend_range?: string | null;
}

export interface StoreUpdate {
  name?: string | null;
  city?: string | null;
  district?: string | null;
  address?: string | null;
  phone?: string | null;
  business_hours?: string | null;
  table_count?: number | null;
  table_types?: string | null;
  pricing?: PricingTier[] | string | null;
  member_cards?: MemberCard[] | string | null;
  operation_profile?: OperationProfile | null;
  has_private_room?: boolean;
  has_coaching?: boolean;
  has_tournament?: boolean;
  has_parking?: boolean;
  target_customers?: string | null;
  style?: string | null;
  brand_style?: string | null;
  advantages?: string | null;
  common_activities?: string | null;
  // Expanded profile fields
  coach_count?: number | null;
  coach_service_types?: string | null;
  coach_price_range?: string | null;
  cue_price_range?: string | null;
  table_brands?: string | null;
  cue_brands?: string | null;
  other_equipment?: string | null;
  membership_types?: unknown | null;
  recharge_rules?: unknown | null;
  membership_benefits?: unknown | null;
  daily_avg_customers?: number | null;
  peak_hours?: string | null;
  avg_spend_range?: string | null;
}

export interface StoreResponse {
  id: string;
  owner_id: string;
  name: string;
  city: string | null;
  district: string | null;
  address: string | null;
  phone: string | null;
  business_hours: string | null;
  table_count: number | null;
  table_types: string | null;
  pricing: PricingTier[] | null;
  member_cards: MemberCard[] | null;
  logo_url: string | null;
  qrcode_url: string | null;
  has_private_room: boolean;
  has_coaching: boolean;
  has_tournament: boolean;
  has_parking: boolean;
  target_customers: string | null;
  style: string | null;
  brand_style?: string | null;
  advantages: string | null;
  common_activities: string | null;
  operation_profile: Record<string, unknown> | null;
  operation_profile_completeness: {
    overall_score: number;
    modules: Record<string, { score: number; completed: boolean; missing_fields: string[] }>;
    completed_modules: string[];
    suggested_modules: string[];
  } | null;
  completeness: number;
  /** 当前用户在本店的角色(owner/manager/...)——工作台默认选中自己的岗位 */
  my_role?: string | null;
  // Expanded profile fields
  coach_count: number | null;
  coach_service_types: string | null;
  coach_price_range: string | null;
  cue_price_range: string | null;
  table_brands: string | null;
  cue_brands: string | null;
  other_equipment: string | null;
  membership_types: unknown | null;
  recharge_rules: unknown | null;
  membership_benefits: unknown | null;
  daily_avg_customers: number | null;
  peak_hours: string | null;
  avg_spend_range: string | null;
  created_at: string;
  updated_at: string;
}


export interface UploadResponse {
  url: string;
}

/** 店脑：门店 AI 记忆条目（「AI 眼里的你的店」页） */
export interface StoreMemoryItem {
  id: string;
  type: string;
  type_label: string;
  content: string;
  confidence: string;
  /** manual=老板亲定的「我的店规矩」 | auto=AI 从使用里学到的 */
  source: "manual" | "auto";
  /** 大白话来源标签：店主定 / AI学到 */
  source_label: string;
}

/** BYOK：门店自带大模型 Key 配置（读，不含明文 key） */
export interface ByokConfigOut {
  enabled: boolean;
  base_url: string | null;
  model: string | null;
  key_configured: boolean;
  key_mask: string;
  image_enabled: boolean;
  image_base_url: string | null;
  image_model: string | null;
  image_key_configured: boolean;
  image_key_mask: string;
  agent_auto_spend_limit: number | null;
}

/** BYOK 多供应商配置档（CC Switch 式：存好几套、一键切换） */
export interface ByokProfile {
  name: string;
  base_url: string | null;
  model: string | null;
  has_key: boolean;
  is_active: boolean;
}

/** BYOK：写配置（api_key 明文，仅提交时传；不传则保留原 key） */
export interface ByokConfigIn {
  enabled: boolean;
  base_url?: string | null;
  api_key?: string | null;
  model?: string | null;
  image_enabled?: boolean;
  image_base_url?: string | null;
  image_api_key?: string | null;
  image_model?: string | null;
  /** 做海报自动出图上限（B-5）：>=0=上限(0=每张先问)；-1=关闭上限闸；不传=不改 */
  agent_auto_spend_limit?: number | null;
}

/** BYOK：测试连接结果 */
export interface ByokValidateResult {
  ok: boolean;
  model?: string;
  sample?: string;
  error?: string;
}

File: server/schemas/store.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, Field


class OperationProfileUpdate(BaseModel):
    """运营画像 — 接受前端发送的嵌套结构，保留所有模块。"""
    basic: dict | None = Field(default=None)
    business_goals: dict | None = Field(default=None)
    customer_structure: dict | None = Field(default=None)
    private_domain_groups: dict | None = Field(default=None)
    assistant_system: dict | None = Field(default=None)
    events: dict | None = Field(default=None)
    commerce_rules: dict | None = Field(default=None)
    equipment: dict | None = Field(default=None)
    content_style: dict | None = Field(default=None)
    ai_preferences: dict | None = Field(default=None)
    # 允许未知键（兼容前端可能增加的模块）
    model_config = {"extra": "allow"}


class StoreCreate(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    city: str | None = Field(default=None, max_length=100)
    district: str | None = Field(default=None, max_length=100)
    address: str | None = Field(default=None, max_length=500)
    phone: str | None = Field(default=None, max_length=50)
    business_hours: str | None = Field(default=None, max_length=200)
    table_count: int | None = Field(default=None, ge=0)
    table_types: str | None = Field(default=None, max_length=500)
    pricing: Any = None
    member_cards: Any = None
    has_private_room: bool = False
    has_coaching: bool = False
    has_tournament: bool = False
    has_parking: bool = False
    target_customers: str | None = Field(default=None, max_length=500)
    style: str | None = Field(default=None, max_length=200)
    advantages: str | None = None
    common_activities: str | None = None
    operation_profile: OperationProfileUpdate | None = None
    # 助教资料
    coach_count: int | None = Field(default=None, ge=0)
    coach_service_types: str | None = Field(default=None, max_length=500)
    coach_price_range: str | None = Field(default=None, max_length=200)
    # 球杆定价
    cue_price_range: str | None = Field(default=None, max_length=200)
    # 设备品牌
    table_brands: str | None = Field(default=None, max_length=500)
    cue_brands: str | None = Field(default=None, max_length=500)
    other_equipment: str | None = None
    # 会员体系
    membership_types: Any = None
    recharge_rules: Any = None
    membership_benefits: Any = None
    # 营业数据
    daily_avg_customers: int | None = Field(default=None, ge=0)
    peak_hours: str | None = Field(default=None, max_length=200)
    avg_spend_range: str | None = Field(default=None, max_length=200)


class StoreUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=200)
    city: str | None = Field(default=None, max_length=100)
    district: str | None = Field(default=None, max_length=100)
    address: str | None = Field(default=None, max_length=500)
    phone: str | None = Field(default=None, max_length=50)
    business_hours: str | None = Field(default=None, max_length=200)
    table_count: int | None = Field(default=None, ge=0)
    table_types: str | None = Field(default=None, max_length=500)
    pricing: Any = None
    member_cards: Any = None
    has_private_room: bool | None = None
    has_coaching: bool | None = None
    has_tournament: bool | None = None
    has_parking: bool | None = None
    target_customers: str | None = Field(default=None, max_length=500)
    style: str | None = Field(default=None, max_length=200)
    # 品牌风格（影响 AI 语气）：此前 schema 缺该字段，前端一直在发但被静默丢弃
    brand_style: str | None = Field(default=None, max_length=50)
    advantages: str | None = None
    common_activities: str | None = None
    operation_profile: OperationProfileUpdate | None = None
    # 助教资料
    coach_count: int | None = Field(default=None, ge=0)
    coach_service_types: str | None = Field(default=None, max_length=500)
    coach_price_range: str | None = Field(default=None, max_length=200)
    # 球杆定价
    cue_price_range: str | None = Field(default=None, max_length=200)
    # 设备品牌
    table_brands: str | None = Field(default=None, max_length=500)
    cue_brands: str | None = Field(default=None, max_length=500)
    other_equipment: str | None = None
    # 会员体系
    membership_types: Any = None
    recharge_rules: Any = None
    membership_benefits: Any = None
    # 营业数据
    daily_avg_customers: int | None = Field(default=None, ge=0)
    peak_hours: str | None = Field(default=None, max_length=200)
    avg_spend_range: str | None = Field(default=None, max_length=200)


class StoreResponse(BaseModel):
    id: UUID
    owner_id: UUID
    name: str
    city: str | None
    district: str | None
    address: str | None
    phone: str | None
    business_hours: str | None
    table_count: int | None
    table_types: str | None
    pricing: Any
    member_cards: Any
    logo_url: str | None
    qrcode_url: str | None
    has_private_room: bool
    has_coaching: bool
    has_tournament: bool
    has_parking: bool
    target_customers: str | None
    style: str | None
    brand_style: str | None = None
    advantages: str | None
    common_activities: str | None
    operation_profile: Any = None
    operation_profile_completeness: Any = None
    completeness: int = 0
    # 本地单用户：恒为 owner（RBAC 多角色已随 SaaS 删除）
    my_role: str | None = None
    # 助教资料
    coach_count: int | None = None
    coach_service_types: str | None = None
    coach_price_range: str | None = None
    # 球杆定价
    cue_price_range: str | None = None
    # 设备品牌
    table_brands: str | None = None
    cue_brands: str | None = None
    other_equipment: str | None = None
    # 会员体系
    membership_types: Any = None
    recharge_rules: Any = None
    membership_benefits: Any = None
    # 营业数据
    daily_avg_customers: int | None = None
    peak_hours: str | None = None
    avg_spend_range: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class StoreListItem(BaseModel):
    """门店列表项（仅 id + name）"""
    id: UUID
    name: str

    model_config = {"from_attributes": True}


class UploadResponse(BaseModel):
    url: str
